### PR TITLE
fix: types in web-pkg

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Fix TypeScript types in web-pkg.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: packages/web-pkg/src/composables/actions/files/useFileActions.ts [hunk 1]
```diff
@@ -224,11 +224,12 @@ export const useFileActions = () => {
   }
 
   const getDefaultAction = (options: GetFileActionsOptions): Action | null => {
-    const filterCallback = (action) =>
+    const filterCallback = (action: FileAction) =>
       action.isVisible({
         ...options,
+        // FIXME: parent does not exit in FileActionOptions, what is this supposed to do?
```
Comment (dschmidt) [inline packages/web-pkg/src/composables/actions/files/useFileActions.ts:230]: uargh

File: packages/web-pkg/src/composables/localStorage/useLocalStorage.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+export const useLocalStorage = <T>(key: string, defaultValue: T = undefined) => {
   const existingValue = window.localStorage.getItem(key)
   const variable = ref(defaultValue)
 
   if (existingValue) {
     try {
       variable.value = JSON.parse(existingValue)
     } catch {
-      variable.value = existingValue
+      ;(variable as Ref<string>).value = existingValue
```
Comment (dschmidt) [inline packages/web-pkg/src/composables/localStorage/useLocalStorage.ts:11]: This doesn't look like it should be necessary or even completely correct

File: packages/web-pkg/tests/unit/components/FilesList/ResourceTable.spec.ts [hunk 3]
```diff
@@ -351,7 +351,9 @@ describe('ResourceTable', () => {
       const tr = await wrapper.find('.oc-tbody-tr-forest .oc-resource-name')
       await tr.trigger('click')
 
-      expect(wrapper.emitted().fileClick[0][0].resources[0].name).toMatch('forest.jpg')
+      expect(
+        (wrapper.emitted('fileClick')[0][0] as { resources: Resource[] }).resources[0].name
```
Comment (dschmidt) [inline packages/web-pkg/tests/unit/components/FilesList/ResourceTable.spec.ts:355]: can we maybe type the emit better? Or is that only possible with script setup?

**Review by dschmidt — APPROVED** [review]
nice! Thanks! 🚀

File: packages/web-pkg/src/composables/localStorage/useLocalStorage.ts (hunk 2 above)
Comment (JammingBen) [reply packages/web-pkg/src/composables/localStorage/useLocalStorage.ts:11]: Problem is that `existingValue` is a string, so in that catch-phrase here `variable` will eventually be a string as well.

File: packages/web-pkg/tests/unit/components/FilesList/ResourceTable.spec.ts (hunk 3 above)
Comment (JammingBen) [reply packages/web-pkg/tests/unit/components/FilesList/ResourceTable.spec.ts]: As discussed, seems only be possible with script setup.

**Pull request merged**